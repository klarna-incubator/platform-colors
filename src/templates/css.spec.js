const generate = require('./css');
const parseColorManifest = require('../parse-color-manifest');
const lightColors = parseColorManifest(require('./fixtures/light-colors.json'));
const darkColors = parseColorManifest(require('./fixtures/dark-colors.json'));
const highContrastColors = parseColorManifest(
  require('./fixtures/high-contrast-colors.json')
);

describe('css template', () => {
  it.each([
    ['light', lightColors],
    ['dark', darkColors],
    ['high contrast', highContrastColors],
  ])('generates %s colors', (name, colors) => {
    expect(generate(colors)).toMatchSnapshot();
  });
});
