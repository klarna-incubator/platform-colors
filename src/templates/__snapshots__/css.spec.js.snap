// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css template generates dark colors 1`] = `
Array [
  Array [
    "colors.css",
    ":root {
  --color-stringColor: #ffffff;
  --color-alphaColor: #ff000080;
}

@media (prefers-color-scheme: dark) {
  :root {
    --color-stringColor: #000000;
    --color-alphaColor: #ffc8c880;
  }
}",
  ],
]
`;

exports[`css template generates high contrast colors 1`] = `
Array [
  Array [
    "colors.css",
    ":root {
  --color-stringColor: #cccccc;
}

@media (prefers-contrast: high) {
  :root {
    --color-stringColor: #ffffff99;
  }
}

@media (prefers-color-scheme: dark) {
  :root {
    --color-stringColor: #33333399;
  }
}

@media (prefers-contrast: high) and (prefers-color-scheme: dark) {
  :root {
    --color-stringColor: #000000;
  }
}",
  ],
]
`;

exports[`css template generates light colors 1`] = `
Array [
  Array [
    "colors.css",
    ":root {
  --color-stringColor: #ffffff;
  --color-alphaColor: #ff000080;
  --color-namedColor: #0000ff;
}",
  ],
]
`;
